### ISSUE TITLE
Documenting code

### ISSUE THREAD

**Issue by c-martinez** [D2KLab/py-sparql-transformer#9]
Hi @pasqLisena,

I'm  running into some issues on how grlc uses [pre_process](https://github.com/D2KLab/py-sparql-transformer/blob/7b2e58b2df1b1d001aabd6b4f4872b2fa95217c5/SPARQLTransformer.py#L38) function. It looks like when I [call the function](https://github.com/CLARIAH/grlc/blob/f0c66780f92bb11c2021feaacf278460d1553fcb/src/utils.py#L241), the returned opt has `'is_json_ld'`  field as False.

I am trying with this query:

```SPARQL
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX dbo: <http://dbpedia.org/ontology/>
PREFIX dbr: <http://dbpedia.org/resource/>
PREFIX dcat: <http://www.w3.org/ns/dcat#>
PREFIX foaf: <http://xmlns.com/foaf/0.1/>

SELECT DISTINCT ?id (sql:BEST_LANGMATCH(?v2, "en;q=1, it;q=0.7 *;q=0.1", "en") AS ?v2) ?v3 ?v4r ?v41
        WHERE {
          ?id a dbo:City.
          ?id dbo:country dbr:Italy.
          ?id rdfs:label ?v2.
          ?id foaf:depiction ?v3.
          ?id dbo:region ?v4r.
          OPTIONAL { ?v4r rdfs:label ?v41.
          FILTER(lang(?v41) = 'it') }
        }
         LIMIT 50 OFFSET 0
```

This relates to grlc issue on [Returning JSON-LD](https://github.com/CLARIAH/grlc/issues/385).

Could you explain what format the function expects?

PS -- I suspect I will run into similar issues with post_process function...

**Comment by c-martinez** [comment]
Am I right to understand that, when the query is a NOT a JSON-LD query, it should include a `proto`  element with the transformation prototype, and that for JSON-LD queries this would be part of the `@graph` element?

**Comment by pasqLisena** [comment]
Yes exactly, that is the discrimination between the two.
So, with `proto` the output is json, with `@graph` is jsonld

**Comment by c-martinez** [comment]
Closed via #10
